app/Livewire/Product/Index.php

```
@@ -2,15 +2,212 @@
+        $query = match ($this->sortBy) {
+            'price_low' => $query->orderBy('regular_price', 'asc'),
+            'price_high' => $query->orderBy('regular_price', 'desc'),
+            'popular' => $query->orderBy('sales_count', 'desc'),
+            'rating' => $query->withAvg('ratings', 'stars')->orderByDesc('ratings_avg_stars'),
+            'latest' => $query->orderBy('publish_date', 'desc'),
+            'oldest' => $query->orderBy('publish_date', 'asc'),
+            default => $query->orderBy('publish_date', 'desc'),
+        };
+
+        return $query->paginate(12);
+    }
```

PHPMD (2.15.0)

The method getProducts() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10. (Code Size Rules)

(CyclomaticComplexity)